test/terraform_aws_nextjs_serverless_test.go

```diff
@@ -0,0 +1,166 @@
+	// fileList = append(fileList, getStaticAssets(aliasURL)...)
+	fileList := getStaticAssets(aliasURL)
+	fmt.Println(fileList)
+
+	test_structure.RunTestStage(t, "staticAssetsCheck", func() {
+		checkIfFilesExist(t, fileList, aliasURL)
+	})
+}
+
+func deployTerraform(t *testing.T, workingDir string, varfiles []string) {
+	terraformOptions := terraform.WithDefaultRetryableErrors(t, &terraform.Options{
+		TerraformDir: workingDir,
+		Reconfigure:  true,
+		VarFiles:     varfiles,
+	})
+	terraform.InitAndApply(t, terraformOptions)
+}
+
+func undeployTerraform(t *testing.T, workingDir string, varfiles []string) {
+	terraformOptions := terraform.WithDefaultRetryableErrors(t, &terraform.Options{
+		TerraformDir: workingDir,
+		Reconfigure:  true,
+		VarFiles:     varfiles,
+	})
+	terraform.Destroy(t, terraformOptions)
+}
+
+// the front page should contain next.js in the raw text and it should respond with 200
+func hitTheFrontPage(t *testing.T, aliasURL string) {
+	tlsConfig := tls.Config{}
+	code, body := http_helper.HttpGet(t, "https://"+aliasURL, &tlsConfig)
+	assert.Equal(t, code, 200)
+	assert.Contains(t, strings.ToLower(body), "next.js")
+}
+
+// hitting icons, which are public assets
+func getIcons(t *testing.T, aliasURL string) {
+	tlsConfig := tls.Config{}
+	code, _ := http_helper.HttpGet(t, "https://"+aliasURL+"/assets/vercel.svg", &tlsConfig)
+	assert.Equal(t, code, 200)
+	code, _ = http_helper.HttpGet(t, "https://"+aliasURL+"/assets/next.svg", &tlsConfig)
+	assert.Equal(t, code, 200)
+	code, _ = http_helper.HttpGet(t, "https://"+aliasURL+"/favicon.ico", &tlsConfig)
+	assert.Equal(t, code, 200)
+}
+
+// Accepts an array of strings as fileList, hits the aliasURL with these paths, expects 200
+func checkIfFilesExist(t *testing.T, fileList []string, aliasURL string) {
+	tlsConfig := tls.Config{}
+	exists := true
+	for _, files := range fileList {
+		fmt.Println("https://" + aliasURL + files)
+		code, _ := http_helper.HttpGet(t, "https://"+aliasURL+files, &tlsConfig)
+		if code == 200 {
+			exists = true
+		} else {
+			exists = false
+			break
+		}
+	}
+	assert.True(t, exists)
+}
+
+// test stage for the files mentioned in required-server-files.json
+func getRequiredServerFiles(aliasURL string) []string {
+	// Locally stored location of this file, we don't expect this to change?
+	requiredServerFiles, _ := os.ReadFile("../examples/nextjs-v13/standalone/.next/required-server-files.json")
+	RSFData := gjson.Parse(string(requiredServerFiles)).Get("files").String()
+	var fileList []string
+	gjson.Parse(RSFData).ForEach(func(_, value gjson.Result) bool {
+		stringValue := value.String()
+		stringValue = "/_" + stringValue[1:]
+		stringValue = strings.ReplaceAll(stringValue, "\\", "/")
+		fileList = append(fileList, stringValue)
+		return true
+	})
+	return fileList
+}
+
+// test stage for the files mentioned in build-manifest.json
+func getStaticAssets(aliasURL string) []string {
```

Comment:
This same is a bit misleading since we're not getting assets but assetURLs. It also contains an unused param aliasURL.

Reply:
I build the full URLs later in the test stage. This function collects the paths, and indeed doesn't use the aliasURL. Fixed.